native_client/README.md

```diff
@@ -1,105 +1,137 @@
-# DeepSpeech native client, language bindings, and custom decoder
+# Building DeepSpeech Binaries
 
-This folder contains the following:
+If you'd like to build the DeepSpeech binaries yourself, you'll need the following pre-requisites downloaded and installed:
 
-1. A native client for running queries on an exported DeepSpeech model
-2. Python and Node.JS bindings for using an exported DeepSpeech model programatically
-3. A CTC beam search decoder which uses a language model (N.B - the decoder is also required for training DeepSpeech)
+* [Mozilla's TensorFlow `r1.13` branch](https://github.com/mozilla/tensorflow/tree/r1.13)
+* [General TensorFlow requirements](https://www.tensorflow.org/install/install_sources)
+* [libsox](https://sourceforge.net/projects/sox/)
+
+It is required to use our fork of TensorFlow since it includes fixes for common problems encountered when building the native client files.
 
-We provide pre-built binaries for Linux and macOS.
+If you'd like to build the language bindings or the decoder package, you'll also need:
+
+* [SWIG](http://www.swig.org/)
+* [node-pre-gyp](https://github.com/mapbox/node-pre-gyp) (for Node.JS bindings only)
 
-## Required Dependencies
 
-Running inference might require some runtime dependencies to be already installed on your system. Those should be the same, whatever the bindings you are using:
-* libsox2
-* libstdc++6
-* libgomp1
-* libpthread
+## Dependencies
 
-Please refer to your system's documentation on how to install those dependencies.
+If you follow these instructions, you should compile your own binaries of DeepSpeech (built on TensorFlow using Bazel).
 
-## Installing our Pre-built Binaries
+For more information on configuring TensorFlow, read the docs up to the end of ["Configure the Build"](https://www.tensorflow.org/install/source#configure_the_build).
 
-To download the pre-built binaries, use `util/taskcluster.py`:
+### TensorFlow: Clone & Checkout
+
+Clone our fork of TensorFlow and checkout the correct version:
 
 ```
-python util/taskcluster.py --target /path/to/destination/folder
+git clone git@github.com:mozilla/tensorflow.git
+git checkout origin/r1.13
 ```
 
-If you need binaries which are different than current master (e.g. `v0.2.0-alpha.6`) you can use the `--branch` flag:
+### TensorFlow: Dependencies
+
+Prepare your environment so you can TensorFlow:
 
-```bash
-python3 util/taskcluster.py --branch "v0.2.0-alpha.6"
+```
+sudo apt install python3-dev python3-pip
+pip install -U --user pip six numpy wheel setuptools mock
+pip install -U --user keras_applications==1.0.6 --no-deps
+pip install -U --user keras_preprocessing==1.0.5 --no-deps
```

Comment:
Don't copy these here, we've already linked to the upstream docs, otherwise we have to keep this file in sync with upstream.

Reply:
removed: 7ab0962b3d31c0d9f89238da2962c2edce998e7b